### ISSUE TITLE
getMetadataKey failed with "Error: Unknown object type 'global' "

### ISSUE THREAD

**Issue by ZeekoZhu** [nestjs/nest#3171]
## Current behavior

When using asciidoctor.js, Nest throws error on startup:

```
Debugger listening on ws://127.0.0.1:9229/89f5305f-ea94-4c87-a7c2-0ee8326b9c77
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
[Nest] 12259   - 10/12/2019, 10:33:45 AM   [NestFactory] Starting Nest application...
[Nest] 12259   - 10/12/2019, 10:33:45 AM   [InstanceLoader] AppModule dependencies initialized +13ms
[Nest] 12259   - 10/12/2019, 10:33:45 AM   [RoutesResolver] TextController {/text}: +4ms
Debugger listening on ws://127.0.0.1:9229/89f5305f-ea94-4c87-a7c2-0ee8326b9c77
For help, see: https://nodejs.org/en/docs/inspector
Error: Unknown object type "global"
    at Object._object (/home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:218:17)
    at Object.dispatch (/home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:185:30)
    at /home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:246:18
    at Array.forEach (<anonymous>)
    at Object._object (/home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:242:21)
    at Object.dispatch (/home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:185:30)
    at /home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:260:23
    at Array.forEach (<anonymous>)
    at Object._array (/home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:259:20)
    at Object._object (/home/zeeko/Projects/text-maid/node_modules/object-hash/index.js:214:30)
```

## Input Code

```ts
import { Injectable } from '@nestjs/common';
import AsciiDoctor from 'asciidoctor.js';
import highlightJsExt from 'asciidoctor-highlight.js';

const asciiDoctor = new AsciiDoctor();
const registry = asciiDoctor.Extensions.create();
highlightJsExt.register(registry);

@Injectable()
export class AsciiDocService {
    render(src: string) {
        const html = asciiDoctor.convert(src);
        return { source: src, html };
    }
}
```

## Expected behavior

No error on startup.

## Possible Solution

I think we can add [`ignoreUnknown: true`](https://www.npmjs.com/package/object-hash#hashvalue-options) option when [generate object hash using object-hash](https://github.com/nestjs/nest/blob/262c50aad511714c75dbf8364ebee690c0ff68d9/packages/core/helpers/handler-metadata-storage.ts#L40).

## Environment

<pre><code>
Nest version: 6.7.2

 
For Tooling issues:
- Node version: v12.11.1
- Platform:  Linux

</code></pre>

**Comment by ZeekoZhu** [comment]
BTW, I don't think using controller instance's object-hash as metadata key is a good idea, sometimes, hashing a large object can be very expensive. 

May be we can generate and attach an uuid to a controller when adding it to a module, then use this uuid as metadata key.

**Comment by kamilmysliwiec** [comment]
Can you please provide a minimal repository which reproduces your issue?

**Comment by ZeekoZhu** [comment]
@kamilmysliwiec  https://github.com/ZeekoZhu/text-maid

```bash
$ git clone https://github.com/ZeekoZhu/text-maid.git
$ cd text-main
$ yarn
$ nest start

[Nest] 6865   - 10/12/2019, 4:29:46 PM   [NestFactory] Starting Nest application...
[Nest] 6865   - 10/12/2019, 4:29:46 PM   [InstanceLoader] AppModule dependencies initialized +9ms
[Nest] 6865   - 10/12/2019, 4:29:46 PM   [RoutesResolver] TextController {/text}: +3ms
Error: Unknown object type "global"
```

**Comment by ZeekoZhu** [comment]
This is caused by https://github.com/opal/opal/issues/2019

**Comment by kamilmysliwiec** [comment]
Fixed in 6.8.4. Let me know if you face any issues.
